sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/utils/SchemaUtil.scala

```diff
@@ -150,8 +264,63 @@ object SchemaUtil {
             .add("partition_id", IntegerType)
         }
 
+      case MapState =>
+        val groupingKeySchema = SchemaUtil.getSchemaAsDataType(
+          stateStoreColFamilySchema.keySchema, "key")
+        val userKeySchema = stateStoreColFamilySchema.userKeyEncoderSchema.get
+        val valueMapSchema = new MapType(
+          keyType = userKeySchema,
+          valueType = stateStoreColFamilySchema.valueSchema,
+          valueContainsNull = false
```

Comment:
Is this required ?

Reply:
Modified as `MapType.apply()` to get rid of assigning `valueContainsNull = false`.